pl_bolts/models/autoencoders/basic_ae/basic_ae_module.py

```diff
@@ -43,24 +43,24 @@ def __init__(
         self.save_hyperparameters()
 
         # link default data
-        if datamodule is None:
-            datamodule = MNISTDataModule(data_dir=self.hparams.data_dir, num_workers=self.hparams.num_workers)
+        # if datamodule is None:
+        #     datamodule = MNISTDataModule(data_dir=self.hparams.data_dir, num_workers=self.hparams.num_workers)
 
-        self.datamodule = datamodule
+        # self.datamodule = datamodule
 
-        self.img_dim = self.datamodule.size()
+        # self.img_dim = self.datamodule.size()
```

Comment:
if we already have `self.hparams.input_x` then why use this? again, makes things harder to follow.

Reply:
I see...its picked up by one of the callbacks. don't want to start changing unrelated things, so leaving it for now.